ast/expressions.go

```diff
@@ -200,7 +210,15 @@ type WhenClause struct {
 
 // Restore implements Node interface.
 func (n *WhenClause) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	ctx.WriteKeyWord(" WHEN ")
```

Comment:
```suggestion
 	ctx.WriteKeyWord("WHEN ")
```

Reply:
We need to comply that the output of `Restore` is a trimed string.If we don't do this, we can only read the code to confirm whether there is a space